File: 2015/8/8.js
var mirps = 0;
for (var i = 13; i < 1000; i += 2) {
	if (!isPalindrome(i) && isPrime(i) && isPrime(reverse(i))) {
		mirps++;
	}
}
console.log(mirps);

function isPrime(num) {
	var sqrt = Math.sqrt(num);
	var isPrime = num % 2 != 0;
	for (var i = 3; i <= sqrt && isPrime; i += 2) {
		isPrime = num % i != 0;
	}
	return isPrime;
};

function isPalindrome(num) {
	return reverse(num) == num;
};

function reverse(num) {
	return parseInt((num + "").split("").reverse().join(""));
};

File: 2015/8/8.hs
isPrime :: [Int] -> Int -> Bool
isPrime (p:ps) n = p*p > n || n `rem` p /= 0 && isPrime ps n

primes :: [Int]
primes = 2:3:5:primes'
    where primes' = 7 : filter (isPrime primes') (scanl (+) 11 $ cycle [2,4,2,4,2,6,2,6])

reverseNum :: Int -> Int
reverseNum = read . reverse . show

isPalindrome :: Int -> Bool
isPalindrome n = n == reverseNum n

isMirp :: Int -> Bool
isMirp = do
    prime <- isPrime primes
    reversePrime <- isPrime primes . reverseNum
    palindrome <- isPalindrome
    return (prime && reversePrime && not palindrome)

main = do
    putStrLn $ show $ length $ filter isMirp [1..999]
